_posts/2020-11-19-testy-e2e-cypress.md

```diff
@@ -0,0 +1,131 @@
+<span class="img-legend">Rys. 2. Struktura projektu Cypress</span>
+</div>
+Objaśnienia:
+- `fixtures` - miejsce gdzie możemy przechowywać gotowe zestaw danych, np. służące do zamockowania usług,
+- `integration` - lokalizacja testów,
+- `plugins` - miejsce do załączania zewnętrznych rozszerzeń dla Cypressa,
+- `support` - tutaj znajdą się np. stałe powtarzające się w wielu scenariuszach czy też nowe, customowe polecenia do globalnego reużycia.
+
+Następnie przy pomocy komendy `npx cypress open` możemy uruchomić dashboard i jeden z przykładowych testów.
+
+# Scenariusze testowe
+W ramach tego wpisu, na warsztat weźmiemy stronę główną bloga Consdata Tech -  [https://blog.consdata.tech/](blog.consdata.tech) i stworzymy dla niej dwa przypadki testowe.
+
+Pierwszy scenariusz będzie polegał na wejście na stronę, poczekaniu aż się załaduje i zweryfikowaniu kilku elementów, która potwierdzą nam, że portal jest w pełni działający.
+
+Natomiast drugi scenariusz, nieco bardziej rozbudowany, dokona kilku interakcji z blogiem.
+
+# Pierwszy test
+
+W katalogu `integration` stwórzmy nowy plik o nazwie `blog.spec.js`, przejdźmy do jego edycji i uzupełnijmy go o następującą treść:
+```javascript
+describe('Blog Consdata Tech', () => {
+    it('should check page fully loaded', () => {
+        // ..
+    });
+})
+```
+Tak jak wcześniej wspomniałem, jeśli ktoś miał wcześniej styczność z testami w JS, wszystko będzie dla niego jasne.
+
+Idąc od początku, `describe` nazywa nam całościowy kontekst, którego będą dotyczyć poszczególne scenariusze. `it` rozpoczyna konkretny przypadek testowy i to wewnątrz niego będziemy implementować właściwą logikę.
+
+W tym momencie możemy uruchomić dashboard Cypressa. Zobaczymy w nim, że pojawił się nasz plik. Po kliknięciu w niego uruchomi się przeglądarka oraz scenariusz testowy:
+<div class="img-with-legend">
+<img alt="Cypress scenariusz testowy" src="/assets/img/posts/2020-11-19-testy-e2e-cypress/scenariusz-testowy-cypress.png" />
+<span class="img-legend">Rys. 3. Uruchomiony scenariusz w Cypress</span>
+</div>
+Zostawmy przeglądarkę i dashboard włączony w tle - Cypress nasłuchuje na zmiany i od razu je wykonuje. Dzięki temu mamy ciągłą pętlę zwrotną z informacją czy to co tworzymy przynosi oczekiwane efekty.
+
+Wracając do tworzenia testu, jak wynika z nazwy `should check page fully loaded`, pierwszy z nich będzie sprawdzał czy strona na którą wchodzimy poprawnie się załadowała. Dodajmy ciało do pustej metody:
+```javascript
+cy.visit('https://blog.consdata.tech/').then(() =>{
+    cy.get('.header-logo').should('be.visible');
+    cy.get('footer').should('be.visible');
+});
+```
+Jak widać, napisany kod jest właściwie samoopisujący się. W pierwszej linii odwiedzamy podany adres, następnie, gdy to się zadzieje, szukamy w DOM elementu o klasie `header-logo` i sprawdzamy czy jest widoczny. Operacje powtarzamy dla elementu `footer`. W przeglądarce powinien uświadczyć nas poniższy obrazek:
```

Comment:
Nie jestem pewien, czy to obrazek powinien uświadczyć czytelnika, czy czytelnik obrazek...

Reply:
zmieniłem lekko